### ISSUE TITLE
Error compiling python bining

### ISSUE THREAD

**Issue by pcyin** [clab/dynet#168]
Hi, I got the following Cython error `Return with value in void function` when installing dynet with python binding:

```
running build_ext
cythoning dynet.pyx to dynet.cpp

Error compiling Cython file:
------------------------------------------------------------
...
        self.thisptr.backward(index)

    cpdef print_graphviz(self):
        self.thisptr.print_graphviz()

    cpdef void checkpoint(self):
         ^
------------------------------------------------------------

dynet.pyx:515:10: Return with value in void function

Error compiling Cython file:
------------------------------------------------------------
...
        self.thisptr.print_graphviz()

    cpdef void checkpoint(self):
        self.thisptr.checkpoint()

    cpdef void revert(self):
         ^
------------------------------------------------------------

dynet.pyx:518:10: Return with value in void function
building '_dynet' extension
/usr/bin/clang -fno-strict-aliasing -fno-common -dynamic -arch i386 -arch x86_64 -g -DNDEBUG -g -fwrapv -O3 -Wall -I/Users/yinpengcheng/Projects/dynet -I/Users/yinpengcheng/Projects/eigen -I/usr/local/include -I/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c dynet.cpp -o build/temp.macosx-10.6-intel-2.7/dynet.o -std=c++11 -stdlib=libc++ -mmacosx-version-min=10.7
dynet.cpp:1:2: error: Do not use this file, it is the result of a failed Cython compilation.
#error Do not use this file, it is the result of a failed Cython compilation.
 ^
1 error generated.
```
Any suggestions? Thanks!

**Comment by neubig** [comment]
I wasn't able to reproduce this compile error, but could you try removing the two "void"s from dynet.pyx?

**Comment by pcyin** [comment]
Problem solved by updating Cython from 0.22 to 0.25.

Meanwhile, I find on mac OS X, the `rpath` might not be set correctly to `~/build/dynet/`, where `libdynet.dylib` exists. This will give error `Library not loaded: @rpath/libdynet.dylib` when `import dynet`. And this could be solved by adding the following line to `.bash_profile`:

`export DYLD_LIBRARY_PATH="$DYLD_LIBRARY_PATH:/path/to/dynet/build/dynet"`

**Comment by pcyin** [comment]
@neubig Issue #146 could be solved by exporting `DYLD_LIBRARY_PATH`.
